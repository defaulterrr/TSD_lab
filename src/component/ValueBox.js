import React from 'react'

class ValueBox extends React.Component {

    constructor(props) {
        super(props)
        this.state = {
            label:  props.label,
            id:     props.id,
            value:  props.value,
            // action: props.action
        }

        this.onTodoChange = this.onTodoChange.bind(this)

    }

    onTodoChange(newvalue){
        this.setState({
            label:  this.props.label,
            id:     this.props.id,
            value:  newvalue
        });
    }

    render() {
        return (
            <div className={this.state.id}>
                <label className="inputLabel">
                    {this.state.label}
                </label>

                <input type="text" className="inputValue" value={this.state.value} id={this.state.id} onChange={e => this.onTodoChange(e.target.value)}>
                    
                </input>
            </div>

        )
    }

    defaultValue = ({value}) => {
        return "value="+value 
    }


}

export default ValueBox;