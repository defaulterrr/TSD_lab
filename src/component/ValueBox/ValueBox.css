.ValueBox {
    margin: 25px;
}