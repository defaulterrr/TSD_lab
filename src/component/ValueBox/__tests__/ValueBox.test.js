import React from 'react'
import { create } from 'react-test-renderer'

import ValueBox from './../ValueBox'

describe("[RENDER] ValueBox â€“ Snapshot test", ()=> {
    test("Matches the snapshot", () => {
        const valuebox = create(<ValueBox label="ValueBox" id="valuebox" value="0"/>)
        expect(valuebox.toJSON()).toMatchSnapshot()
    })
})