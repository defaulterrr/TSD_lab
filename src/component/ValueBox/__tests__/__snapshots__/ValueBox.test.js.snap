// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RENDER] ValueBox â€“ Snapshot test Matches the snapshot 1`] = `
<div
  class="ValueBox"
  className="valuebox"
>
  <label
    className="inputLabel"
  >
    ValueBox
  </label>
  <input
    className="inputValue"
    id="valuebox"
    onChange={[Function]}
    type="text"
    value="0"
  />
</div>
`;
